src/Application/Services/ItemService.cs

```diff
@@ -39,86 +46,97 @@ public async Task<Item> Get(string key)
-
-            ItemEntity itemEntity = new()
+            Item item = new()
             {
-                Key = itemDto.Key,
-                Value = JsonSerializer.Serialize(itemDto.Value),
-                ExpirationPeriod = ExPeriod,
-                ExpirationDate = DateTime.Now.AddSeconds(ExPeriod)
+                Key = itemEntity.Key,
+                Value = JsonSerializer.Deserialize<List<string>>(itemEntity.Value),
+                ExpirationPeriod = itemEntity.ExpirationPeriod,
+                ExpirationDate = updatedDate
             };
 
-            await _itemRepository.Create(itemEntity);
+            await _itemRepository.UpdateExDate(itemEntity.Key, updatedDate);
 
-            return "Item created";
+            return item;
         }
         else
         {
-            int ExPeriod = itemDto.ExpirationPeriod ?? 0;
+            await _itemRepository.Delete(key);
 
-            ItemEntity itemEntity = new()
+            throw new NotFoundException();
+        }
+
+    }
+
+    public async Task<string> Create(ItemCreate itemDto)
+    {
+        if (itemDto.ExpirationPeriod < MaxItemExPeriod)
+        {
+            if (await _itemRepository.Get(itemDto.Key) is null)
             {
-                Key = itemDto.Key,
-                Value = JsonSerializer.Serialize(itemDto.Value),
-                ExpirationPeriod = ExPeriod,
-                ExpirationDate = DateTime.Now.AddSeconds(ExPeriod)
-            };
+                int exPeriod = itemDto.ExpirationPeriod is 0 ? DefaultExPeriod : itemDto.ExpirationPeriod.Value;
+
+                ItemEntity itemEntity = new()
+                {
+                    Key = itemDto.Key,
+                    Value = JsonSerializer.Serialize(itemDto.Value),
+                    ExpirationPeriod = exPeriod,
+                    ExpirationDate = DateTime.Now.AddSeconds(exPeriod)
+                };
+
+                await _itemRepository.Create(itemEntity);
 
-            await _itemRepository.Create(itemEntity);
+                return "Item created";
+            }
 
-            return "Item created";
+            throw new FoundException("Item with same key already exists.");
         }
+
+        throw new ArgumentException($"Invalid ExpirationPeriod field");
     }
 
     public async Task<string> Update(ItemCreate itemDto)
     {
-        if (await _itemRepository.Get(itemDto.Key) is null)
+        if (itemDto.ExpirationPeriod < MaxItemExPeriod)
         {
-            int ExPeriod = itemDto.ExpirationPeriod ?? 0;
-
-            ItemEntity itemEntity = new()
+            if (await _itemRepository.Get(itemDto.Key) is null)
             {
-                Key = itemDto.Key,
-                Value = JsonSerializer.Serialize(itemDto.Value),
-                ExpirationPeriod = ExPeriod,
-                ExpirationDate = DateTime.Now.AddSeconds(ExPeriod)
-            };
+                int exPeriod = itemDto.ExpirationPeriod is 0 ? DefaultExPeriod : itemDto.ExpirationPeriod.Value;
```

Comment:
incorrect code, should be "If expiration period is not defined - default value should be used from config"